### DISCUSSION TITLE
Better error handling to improve developer experience

### DISCUSSION THREAD

**Discussion by bytekai** [alpinejs/alpine#3555] [3. Feature ideas]
When using the x-for directive and accidentally using a key that does not exist, instead of throwing a helpful error message saying that the key that is being used does not exist we get the typical `undefined trying to access 'after'`.

I'm unsure if no null checking/error handling is common for this code base but i believe that good error handling can greatly improve the developer experience.

**Comment by ekwoka** [comment]
Most of the time, errors in expressions go through an error handling process to provide more information.

a bad parse on the for loop does not similarly happen for some reason.

**Comment by Saoming** [comment]
+1 for this.

I agree with you @ekwoka. I used the  X-FOR yesterday to parse rest_api objects inside my WordPress Site. However, the parse gives me **undefined trying to access 'after'** The error was: key ID was not uppercase but lowercase however Alpine didn't tell me straight away that the key is the one who make a mistake and I ended up reviewing my store and review my code. wasted so much time.

I think we need to have a better error handling  I am going to create an issue under this discussion.
